Content/posts/jwtkit-v5.md

```diff
@@ -0,0 +1,182 @@
+---
+date: 2024-02-22 16:00
+description: We've released a new version of JWTKit, and it's a big one!
+tags: jwt, security, vapor
+author: Paul; Tim
+authorImageURL: /author-images/paul.jpg
+--- 
+# JWTKit is no longer Boring!
+
+## Swift 6 is on the Horizon
+
+If you follow the Swift forums carefully you might have noticed [the announcement](https://forums.swift.org/t/progress-toward-the-swift-6-language-mode/68315/33) that Swift 5.10 will be the last release before Swift 6. This provides Vapor a timeline for a future Vapor 5 release and we can start planning as to what that will look like.
+
+A large part of that will be updating and migrating all of our packages to use modern Swift features and paradigms like `Sendable`, actors and making use of new APIs. The first package to be updated is JWTKit, which has been in the works for a while now.
+
+A future post will discuss Vapor 5, so let's have a look at what's new in JWTKit.
+
+## JWTKit V5
+
+During the last few months you might have noticed an open pull request in the JWTKit repository called "V5". As you have probably already guessed, this pull request brings a new version, namely number 5, to JWTKit. Until version 4, our beloved JWT library was based mostly on a vendored copy of BoringSSL, a cryptographic library written in plain old C. While it did _work_, maintaining a wrapper around C is not modern anymore, let alone "Swifty". It introduced potential safety issues calling through to C code and could add a significant amount to your app's compile time, when Swift Crypto already compiles another copy! That's why we decided to eradicate BoringSSL from JWTKit and replace it with Swift-only internals, namely Apple's SwiftCrypto. Available as of today in a beta version is major release 5 for JWTKit.
+
+Besides removing the C-based internals, the package got a number of upgrades. JWTKit is now fully `Sendable` and builds without warning with strict concurrency settings. 
+
+We also added a new signing algorithm: `PSS` padded `RSA`. While `RSA` is not recommended anymore, having to use RSA with `PKCS1-v1_5` is even worse, that's why we added the (slightly) safer (but still not safe enough) version of the key. If you want to try it but can't find it, it's likely because the `RSA` key is now gated behind the `Insecure` namespace to discourage new users from using it. 
```

Comment:
```suggestion
We also added a new signing algorithm: `PSS`-padded `RSA` (aka `RSA-PSS`). Although `RSA` is no longer considered future-proof and should be avoided whenever possible, using RSA with the much older PKCS#1 v1.5 padding is _known_ to be unsafe today - so we added support for the safer (though still not recommended) padding mode. If you want to try it but can't find it, it's likely because `RSA` key types are now nested within the `Insecure` namespace, to discourage new users from using it. 
```